### DISCUSSION TITLE
Calendar Component

### DISCUSSION THREAD

**Discussion by vhscom** [saadeghi/daisyui#633] [Ideas / feature requests]
I saw in a few of the issues there may be a calendar component on the horizon. If we're blessed with such a thing, I just want to express how delightful it would be to have it built with consideration for [Full Calendar](https://github.com/fullcalendar/fullcalendar/issues/470) ([website](https://fullcalendar.io/)). There's an existing SvelteKit example of Full Calendar running in the [svelte-fullcalendar](https://github.com/YogliB/svelte-fullcalendar) repo if you're looking for a canvas to play around in. Cheers and looking forward to seeing all the wonderful and creative things this library produces as it moves forward.

**Comment by saadeghi** [comment]
Thanks for the details.  
My main concern is how useful the new `calendar` would be because it would be just the layout and there will be no interactive functionalities by default (since daisyUI is just a Tailwind CSS plugin and there are no JS bundle file included).  
There are many calendar or date picker packages out there. some allow CSS modifications, some don't.  
I think if I just make a layout to show the current day in a monthly view (like this 👇), that wouldn't be useful.  

![de0cb7f02ea7cc2abc5d0e322cd31958](https://user-images.githubusercontent.com/7342023/159208486-d09f96aa-d343-4777-bce0-ea9f7d318bf3.jpg)

people still can handle many things using JS (if day is clicked, give it an `active` class) but is this really worth it to recreate this functionality when there are date pickers and calendar packages with much more functionalities? 

I can make something like this 👇 using checkboxes for each day. but then it will be limited to selecting one day (not a range of days) and developer still needs to give it the calendar data (HTML elements for each day. that would be like ~40 divs with ~30 checkboxes to show a month layout)
![calendar](https://user-images.githubusercontent.com/7342023/159208709-576e2651-11da-4768-be33-cb7fea0272c5.gif)

Something like this 👇 would be useful but then again, it needs a JS bundle.
![header-calendly](https://user-images.githubusercontent.com/7342023/159208979-32080825-5e48-4dc8-8b57-8485c69ccb64.jpg)

And something like Full Calendar would't be helpful if it's just layout without functionalities.

I'm not really sure how this should go.

**Reply by vhscom** [reply]
Date pickers seem to be a dime a dozen these days. And given there's native date pickers which look quite smashing on say, for example, [EMUI](https://consumer.huawei.com/en/emui/), sticking to something which could allow users to lay down some semantic, ARIA-friendly and potentially Schema.org-oriented dates on a static calendar or timeline view might be the sweet spot. Then the user can decide if they want to use it to roll their own Headless UI component, integrate a calendaring library such as FullCalendar or just do something simple like show their vacation journey or wedding schedule. Something like that.

Worth pointing to since we're on the subject of Tailwind:
https://tailwindui.com/components/application-ui/data-display/calendars

**Reply by saadeghi** [reply]
Ok then, I guess a layout-only calendar would be enough to add, and it would be up to the developer on how the handle the functionality.

**Reply by vhscom** [reply]
That's the gist of it. 🙂  Of course I've seen so many cool things coming out of this library (like the swap) that making a calendar might not be particularly interesting. Though I did see several users were looking for calendar/date related functionality so I thought I'd start a dialog. Looking forward to seeing what sorts of awesome stuff you cook up next, Pouya. Cheers. 🥇

**Reply by joeblau** [reply]
I just wanted to chime in and say I was also looking for a layout-only calendar. I have some data that I want to visualize for users in a calendar format but users only need to see the date of the event.

Really appreciate everything you've done so far! Amazing Library.

**Comment by scootcho** [comment]
For Components that need interaction using JS, maybe we can make them into Web Components? The JS will be encapsulated within the Web Component without a JS bundle. And they'll be framework agnostic that can be used by different frontend frameworks (Svelte, Vue, React, etc.)

**Reply by saadeghi** [reply]
There are a few datepicker web components already. Some of them provide styling with their own CSS variables (which we can set daisyUI colors to them)
